File: hardware/src/main/java/com/greenaddress/greenapi/HWWalletBridge.java
package com.greenaddress.greenapi;

import androidx.annotation.Nullable;

import com.blockstream.DeviceBrand;
import com.blockstream.gdk.data.Network;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

public interface HWWalletBridge {
    void interactionRequest(final HWWallet hw, @Nullable final Completable completable, @Nullable final String text);

    String requestPinMatrix(DeviceBrand deviceBrand);
    String requestPassphrase(DeviceBrand deviceBrand);
}

File: hardware/src/main/java/com/greenaddress/greenapi/HWWalletBridge.kt
package com.greenaddress.greenapi

import com.blockstream.DeviceBrand
import kotlinx.coroutines.CompletableDeferred

interface HWWalletBridge {
    fun interactionRequest(hw: HWWallet?, completable: CompletableDeferred<Boolean>?, text: String?)
    fun requestPinMatrix(deviceBrand: DeviceBrand?): String?
    fun requestPassphrase(deviceBrand: DeviceBrand?): String?
}